corfudb-tools/src/main/java/org/corfudb/compactor/CompactorController.java

```diff
@@ -40,12 +42,17 @@ public CompactorController(CompactorControllerConfig config, CorfuRuntime corfuR
      * @param args command line argument strings
      */
     public static void main(String[] args) {
+        CompactorController corfuStoreCompactorControls = null; 
         try {
-            CompactorController corfuStoreCompactorControls = new CompactorController(args);
+            corfuStoreCompactorControls = new CompactorController(args);
             corfuStoreCompactorControls.doCompactorAction();
         } catch (Exception e) {
             log.error("CorfuStoreCompactorMain crashed with error: {}, Exception: ",
                     CompactorBaseConfig.CORFU_LOG_CHECKPOINT_ERROR, e);
+        } finally {
+            if (corfuStoreCompactorControls != null) {
+                corfuStoreCompactorControls.shutdown();
+            }
```

Comment:
You have a similar issue here - If `corfuRuntime` succeeds in connecting, but an exception is thrown during the allocation of `DistributedCheckpointerHelper` when initializing `CompactorMetadataTables`, resources will still be leaked.

Reply:
Be mindful of when `connect()` throws an exception. If it throws an exception after `sslCertWatcher` is initialized, your reference to `corfuRuntime` will still be null, so relying on a null-check here will not suffice with how the code is currently written.